src/cascadia/UnitTests_TerminalCore/ScreenSizeLimitsTest.cpp

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "precomp.h"
+#include <WexTestClass.h>
+#include <climits>
+
+#include "../cascadia/TerminalCore/Terminal.hpp"
+#include "../renderer/inc/DummyRenderTarget.hpp"
+#include "consoletaeftemplates.hpp"
+
+#include "winrt/Microsoft.Terminal.Settings.h"
+
+using namespace winrt::Microsoft::Terminal::Settings;
+using namespace Microsoft::Terminal::Core;
+
+namespace TerminalCoreUnitTests
+{
+class MockTermSettings : public winrt::implements<MockTermSettings, ICoreSettings>
+{
+public:
+    MockTermSettings(int32_t historySize, int32_t initialRows, int32_t initialCols) :
+        _historySize(historySize),
+        _initialRows(initialRows),
+        _initialCols(initialCols)
+    { }
+
+    // property getters - all implemented
+    int32_t HistorySize() { return _historySize; }
+    int32_t InitialRows() { return _initialRows; }
+    int32_t InitialCols() { return _initialCols; }
+    uint32_t DefaultForeground() { return 0xFFFFFFFFUL; }

```

Comment:
_super nit_: a lot of these properties have default values defined in `src/inc/DefaultSettings.h` - you may want to replace them with the named constants from that file.